lib/widgets/checkbox_widget.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:todo_app/screens/detail_page.dart';
+import 'package:todo_app/src/constants.dart';
+import 'package:todo_app/src/models/note.dart';
+import 'package:todo_app/src/provider/provider_data.dart';
+import 'package:todo_app/src/services.dart';
+
+///
+class CheckBoxWidget extends StatefulWidget {
+  ///
+  final Note note;
+
+  ///
+  const CheckBoxWidget({
+    Key? key,
+    required this.note,
+  }) : super(key: key);
+
+  @override
+  State<CheckBoxWidget> createState() => _CheckBoxState();
+}
+
+class _CheckBoxState extends State<CheckBoxWidget> {
+  List<Todo> taskList = [];
```

Comment:
Also -- explore how Map and List behave when they're `final`.
```suggestion
  final List<Todo> _taskList = [];
```